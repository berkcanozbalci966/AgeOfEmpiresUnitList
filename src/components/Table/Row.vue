<template>
  <tr>
    <td>{{ unitInfo.id }}</td>
    <td>{{ unitInfo.name }}</td>
    <td>{{ unitInfo.age }}</td>
    <td>{{ unitInfo.cost }}</td>
  </tr>
</template>

<script>
export default {
  props: {
    unitInfo: {
      type: Object,
      default: () => ({}),
    },
  },
  mounted() {
    console.log(this.unitInfo);
  },
};
</script>

<style></style>
