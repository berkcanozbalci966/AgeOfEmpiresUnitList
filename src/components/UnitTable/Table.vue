<template>
  <table>
    <tr v-if="getUnitInformations.id">
      <th scope="row">ID:</th>
      <td>{{ getUnitInformations.id }}</td>
    </tr>
    <tr v-if="getUnitInformations.name">
      <th scope="row">Name:</th>
      <td>{{ getUnitInformations.name }}</td>
    </tr>
    <tr v-if="getUnitInformations.description">
      <th scope="row">Description:</th>
      <td>{{ getUnitInformations.description }}</td>
    </tr>
    <tr v-if="getUnitInformations.age">
      <th scope="row">Min.Required Age:</th>
      <td>{{ getUnitInformations.age }}</td>
    </tr>
    <template v-if="getUnitInformations.cost">
      <tr v-if="getUnitInformations.cost.Wood">
        <th scope="row">Wood Cost:</th>
        <td>{{ getUnitInformations.cost.Wood }}</td>
      </tr>
      <tr v-if="getUnitInformations.cost.Gold">
        <th scope="row">Gold Cost:</th>
        <td>{{ getUnitInformations.cost.Gold }}</td>
      </tr>

      <tr v-if="getUnitInformations.cost.Food">
        <th scope="row">Food Cost:</th>
        <td>{{ getUnitInformations.cost.Food }}</td>
      </tr>
    </template>

    <tr v-if="getUnitInformations.build_time">
      <th scope="row">Build Time:</th>
      <td>{{ getUnitInformations.build_time }}</td>
    </tr>
    <tr v-if="getUnitInformations.reload_time">
      <th scope="row">Reload Time:</th>
      <td>{{ getUnitInformations.reload_time }}</td>
    </tr>
    <tr v-if="getUnitInformations.hit_points">
      <th scope="row">Hit Points:</th>
      <td>{{ getUnitInformations.hit_points }}</td>
    </tr>
    <tr v-if="getUnitInformations.attack">
      <th scope="row">Attack:</th>
      <td>{{ getUnitInformations.attack }}</td>
    </tr>
    <tr v-if="getUnitInformations.accuracy">
      <th scope="row">Accuracy:</th>
      <td>{{ getUnitInformations.accuracy }}</td>
    </tr>
  </table>
</template>

<script>
export default {
  props: {
    getUnitInformations: {
      type: Object,
      default: () => {},
    },
  },
};
</script>

<style lang="scss" scoped>
caption {
  padding: 10px;
  caption-side: bottom;
}
table {
  border-collapse: collapse;
  border: 2px solid rgb(200, 200, 200);
  letter-spacing: 1px;
  font-family: sans-serif;
  font-size: 0.8rem;
  max-width: 450px;
}
td,
th {
  border: 1px solid rgb(190, 190, 190);
  padding: 7px 5px;
}
th {
  background-color: rgb(235, 235, 235);
}
td {
  text-align: left;
}
tr:nth-child(even) td {
  background-color: rgb(250, 250, 250);
}
tr:nth-child(odd) td {
  background-color: rgb(240, 240, 240);
}
</style>
