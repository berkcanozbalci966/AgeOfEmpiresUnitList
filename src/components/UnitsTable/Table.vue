<template>
  <div>
    <table>
      <thead>
        <tr>
          <th>id</th>
          <th>name</th>
          <th>age</th>
          <th>cost</th>
        </tr>
      </thead>
      <tbody>
        <Row
          v-for="(unitInfo, index) in unitsData"
          :unitInfo="unitInfo"
          :key="index"
        />
      </tbody>
    </table>
  </div>
</template>

<script>
import Row from "@/components/UnitsTable/Row";
export default {
  components: {
    Row,
  },
  props: {
    unitsData: {
      type: Array,
      default: () => [],
    },
  },
};
</script>

<style lang="scss">
table {
  margin-top: 50px;
  width: 100%;
  border-collapse: collapse;
  tr {
    cursor: unset;
    td,
    th {
      border: 1px solid #dddddd;
      text-align: left;
      padding: 8px;
    }
  }
}
</style>
