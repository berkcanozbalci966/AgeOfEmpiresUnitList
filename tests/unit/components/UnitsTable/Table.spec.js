import { mount } from "@vue/test-utils";
import Table from "../../../../src/components/UnitsTable/Table";

describe("Table", () => {
  it("list render", () => {
    const component = mount(Table, {
      props: {
        unitsData: [
          {
            name: "Archer",
            age: "Feudal",
            id: 1,
            cost: "Wood:25, Gold:45",
          },
          {
            name: "Crossbowman",
            age: "Castle",
            id: 2,
            cost: "Wood:25, Gold:45",
          },
          {
            name: "Arbalest",
            age: "Imperial",
            id: 3,
            cost: "Wood:25, Gold:45",
          },
        ],
      },
    });

    expect(component.html()).toMatchSnapshot();
  });
});
