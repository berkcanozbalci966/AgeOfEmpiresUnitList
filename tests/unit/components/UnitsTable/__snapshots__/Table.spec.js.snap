// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table list render 1`] = `
<div>
  <table>
    <thead>
      <tr>
        <th>id</th>
        <th>name</th>
        <th>age</th>
        <th>cost</th>
      </tr>
    </thead>
    <tbody>
      <tr data-test="unitsrow">
        <td>1</td>
        <td>Archer</td>
        <td>Feudal</td>
        <td>Wood:25, Gold:45</td>
      </tr>
      <tr data-test="unitsrow">
        <td>2</td>
        <td>Crossbowman</td>
        <td>Castle</td>
        <td>Wood:25, Gold:45</td>
      </tr>
      <tr data-test="unitsrow">
        <td>3</td>
        <td>Arbalest</td>
        <td>Imperial</td>
        <td>Wood:25, Gold:45</td>
      </tr>
    </tbody>
  </table>
</div>
`;
